File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        List<Integer> list = new ArrayList<>();
        Queue<Integer> q = new LinkedList<>();

        for (int i = 0; i < progresses.length; i++) {
            q.offer(progresses[i]);
        }

        int idx = 0;
        int days = 0;

        while (!q.isEmpty()) {
            double progress = progresses[idx];
            double speed = speeds[idx];

            // days += (100 - progress) % speed == 0 ?
            //     (100 - progress) / speed
            //     : (100 - progress) / speed + 1;
            days = (int) Math.ceil((100 - progress) / speed);

            int completedTasks = 0;
            while (!q.isEmpty() && q.peek() + speeds[idx] * days >= 100) {
                completedTasks++;
                q.poll();
                idx++;
            }

            if (completedTasks > 0) {
                list.add(completedTasks);
            }
        }

        int[] answer = new int[list.size()];
        for (int i = 0; i < answer.length; i++) {
            answer[i] = list.get(i);
        }

        return answer;
    }
}

File: 프로그래머스/2/42586. 기능개발/기능개발.cpp
#include <string>
#include <vector>
#include <queue>

using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;

    int restProgressCount = progresses.size();
    while(restProgressCount) {
        //하루 진행률 합산
        for(int i = 0; i < progresses.size(); i++) {
            progresses[i] += speeds[i];
        }


        int returnCount = 0;
        for(int i = 0; i < progresses.size(); i++) {
            if(progresses[0] < 100) { break; }

            progresses.erase(progresses.begin() + 0);
            speeds.erase(speeds.begin() + 0);
            returnCount++;
            restProgressCount--;
            i--;
        }

        if(returnCount > 0) {
            answer.push_back(returnCount);
        }
    }

    return answer;
}